rpcs3/Emu/RSX/Overlays/HomeMenu/overlay_home_menu.cpp

```diff
@@ -0,0 +1,220 @@
+#include "stdafx.h"
+#include "overlay_home_menu.h"
+#include "Emu/RSX/RSXThread.h"
+
+#include <sstream>
+#include <iomanip>
+
+namespace rsx
+{
+	namespace overlays
+	{
+		std::string get_time_string()
+		{
+			std::ostringstream ost;
+			const std::time_t dateTime = std::chrono::system_clock::to_time_t(std::chrono::system_clock::now());
+ 			const std::tm tm = *std::localtime(&dateTime);
+			ost << std::put_time(&tm, "%Y/%m/%d %H:%M:%S");
+			return ost.str();
+		}
+
+		home_menu_dialog::home_menu_dialog()
+		{
+			m_allow_input_on_pause = true;
+
+			m_dim_background = std::make_unique<overlay_element>();
+			m_dim_background->set_size(overlay::virtual_width, overlay::virtual_height);
+			m_dim_background->back_color.a = 0.5f;
+
+			m_main_menu = std::make_unique<home_menu_main_menu>(20, 85, 1240, 540, false, nullptr);
+
+			m_description = std::make_unique<label>();
+			m_description->set_font("Arial", 20);
+			m_description->set_pos(20, 37);
+			m_description->set_text(m_main_menu->title);
+			m_description->auto_resize();
+			m_description->back_color.a = 0.f;
+
+			m_time_display = std::make_unique<label>();
```

Comment:
Are these items optional? If not, they do not need to be pointers at all. You can just create/store them as they have a trivial ctor

Reply:
fixed